pinot-tools/src/main/java/org/apache/pinot/tools/tuner/meta/manager/collector/CompressedFilePathIter.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tools.tuner.meta.manager.collector;
+
+import io.vavr.Tuple2;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import org.apache.pinot.tools.tuner.query.src.QuerySrc;
+import org.apache.pinot.tools.tuner.query.src.stats.wrapper.AbstractQueryStats;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The iterator over all tarred segments in _directory/tableName/
+ */
+public class CompressedFilePathIter implements QuerySrc {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CompressedFilePathIter.class);
+  Iterator<Tuple2<String, File>> _iterator;
+
+  private String _directory;
+
+  private CompressedFilePathIter(Builder builder) {
+    _directory = builder._directory;
+  }
+
+  private CompressedFilePathIter openDirectory() {
+    File dir = new File(_directory);
+    if (!dir.exists() || dir.isFile()) {
+      LOGGER.error("Wrong input directory!");
```

Comment:
can we bubble this up to Driver/Command class instead of System.exit (for all similar usages), and handle the exceptions in that entry level class?